acapy_agent/revocation/routes.py

```diff
@@ -1784,11 +1784,16 @@ async def delete_tails(request: web.BaseRequest) -> json:
             return web.json_response({"message": str(e)})
     elif cred_def_id:
         async with session:
-            cred_reg = sorted(
+            records = sorted(
                 await IssuerRevRegRecord.query_by_cred_def_id(
                     session, cred_def_id, IssuerRevRegRecord.STATE_GENERATED
                 )
-            )[0]
+            )
+            if not records:
+                return web.json_response(
+                    {"message": "No tail files found for deletion"}, status=404
```

Comment:
You could use the http class via `http.HTTPStatus.NOT_FOUND`. We haven't really done this generally for the project though so this is fine. 

Just mentioning it because I think we should have a task to change it throughout the codebase.